### ISSUE TITLE
The maxium number of file allowed in sync

### ISSUE THREAD

**Issue by focux** [Azure/azure-storage-azcopy#540]
##### Note: The version is visible when running AzCopy without any argument
10.1.2
### Which platform are you using? (ex: Windows, Mac, Linux)
Mac

### What problem was encountered?
```
0 Files Scanned at Source, 9998407 Files Scanned at Destination
Cannot perform sync due to error: the maximum number of file allowed in sync is: 10000000
```
Is there a way to edit the max number of files allowed in sync? I'm trying to sync around 7m of photos, each photo have a size of 50kb.

**Comment by JohnRusk** [comment]
There's not a way to edit it right now. 

Is it possible for you to break the job up - e.g. if your root directory has, say, 10 direct children, could you sync each child separately?  (If you try that, watch out for how your formulate the destination directory/URL. You'll probably want to include the child dir name at the end of both the source and the destination.  Try a test on an empty container first, I'd suggest!).

If it's not practical for you to break the job up, then I'm sorry I don't think we have any other workarounds right now.   cc @zezha-msft for comments and maybe thoughts on whether we should change something in some future release.

**Comment by zezha-msft** [comment]
Hi @focux, we are really sorry for the inconvenience. The 10 million file limitation cannot be easily overwritten right now, it was an arbitrary large number that was picked because we wanted to put an upper bound on how much memory we use to do the comparison of source vs destination. 

I see that there's a need to make this value configurable, or simple remove it. I'll add an item to address this issue.

**Comment by zezha-msft** [comment]
This has been fixed in [10.3.0](https://github.com/Azure/azure-storage-azcopy/releases/tag/v10.3.0).
